File: dotnet/windntrees.net/DataAccess/DBFModels/Metadata/AdvertisementLocation.cs
using SharedLibrary.Attributes;
using System.ComponentModel.DataAnnotations;

namespace DataAccess
{
    [MetadataType(typeof(AdvertisementLocationMD))]
    public partial class AdvertisementLocation
    {
    }

    public partial class AdvertisementLocationMD
    {
        [Key]
        public System.Guid UID { get; set; }

        [LocaleMessageRequired]
        [LocaleMessageStringLength(50)]
        [Display(ResourceType = typeof(LocaleResources.Contents.Category), Name = "Name")]
        public string Name { get; set; }

        [LocaleMessageRequired]
        [LocaleMessageStringLength(255)]
        [Display(ResourceType = typeof(LocaleResources.Contents.Category), Name = "Description")]
        public string Description { get; set; }

        [Timestamp]
        public byte[] RowVersion { get; set; }
    }
}

File: dotnet/windntrees.core/Application.Core/wwwroot/js/models/AdvertisementLocation.js
function AdvertisementLocation(data) {
    var instance = this;

    instance._datakey = data.Uid;

    instance.Uid = data.Uid;
    
    instance.Name = ko.observable(data.Name);
    instance.Name.extend({
        required: true,
        maxLength: 50
    });

    instance.Description = ko.observable(data.Description);
    instance.Description.extend({
        maxLength: 255
    });

    instance.RowVersion = data.RowVersion;
}